code/modules/antagonists/eldritch_cult/eldritch_effects.dm

```diff
@@ -0,0 +1,181 @@
+/obj/effect/eldritch
+	name = "Generic rune"
+	desc = "Weird combination of shapes and symbols etched into the floor itself. The indentation is filled with thick black tar-like fluid."
+	anchored = TRUE
+	icon_state = ""
+	resistance_flags = FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	layer = SIGIL_LAYER
+
+/obj/effect/eldritch/attack_hand(mob/living/user)
+	. = ..()
+	if(.)
+		return
+	try_activate(user)
+
+/obj/effect/eldritch/proc/try_activate(mob/living/user)
+	if(!user.mind.has_antag_datum(/datum/antagonist/e_cult))
+		return
+	flick("[icon_state]_active",src)
+	activate(user)
+
+/obj/effect/eldritch/attacked_by(obj/item/I, mob/living/user)
+	. = ..()
+	if(istype(I,/obj/item/nullrod))
+		qdel(src)
+
+/obj/effect/eldritch/proc/activate(mob/living/user)
+
+	var/datum/antagonist/e_cult/cultie = user.mind.has_antag_datum(/datum/antagonist/e_cult)
+	var/list/knowledge = cultie.get_all_knowledge()
+	var/list/atoms_in_range = list()
+
+	for(var/atom/A in range(1,src))
+		if(istype(A,/mob/living))
+			var/mob/living/L = A
+			if(L.stat != DEAD || L == user) // we only accept corpses, no living beings allowed.
+				continue
+		atoms_in_range += A
+
+	for(var/X in knowledge)
+		var/datum/eldritch_knowledge/EK = X
+		var/list/local_required_atoms = EK.required_atoms
+		if(!local_required_atoms || local_required_atoms.len == 0)
+			continue
+
+		var/list/selected_atoms = list()
+
+		if(!EK.recipe_snowflake_check(atoms_in_range,drop_location(),selected_atoms))
+			continue
+
+		for(var/X1 in local_required_atoms)
+			local_required_atoms[X1] = FALSE
+
+		for(var/X1 in atoms_in_range)
+			var/atom/atom_x1 = X1
+			for(var/X2 in local_required_atoms)
+				var/list/temp_list_X2 = X2
+				if(!is_type_in_list(atom_x1,temp_list_X2) || local_required_atoms[X2] == TRUE)
+					continue
+				local_required_atoms[X2] = TRUE
+				selected_atoms |= atom_x1
+
+		if(selected_atoms.len == 0)
+			continue
+		var/skip = FALSE
+		for(var/X1 in local_required_atoms)
+			if(local_required_atoms[X1] != TRUE)
+				skip = TRUE
+				break
+
+		if(skip)
+			continue
+
+		EK.on_finished_recipe(user,selected_atoms,loc)
+		EK.cleanup_atoms(selected_atoms)
+	return
+
+/obj/effect/eldritch/big
+	name = "Transmutation rune"
+	icon = 'icons/effects/96x96.dmi'
+	icon_state = "eldritch_rune1"
+	pixel_x = -32 //So the big ol' 96x96 sprite shows up right
+	pixel_y = -32
+
+/**
+  * #Reality smash tracker
+  *
+  * Stupid fucking list holder, DONT create new ones, it will break the game, this is automnatically created whenever eldritch cultists are created.
+  *
+  * Tracks relevant data, generates relevant data, useful tool
+  */
+/datum/reality_smash_tracker
+	///list of tracked reality smashes
+	var/list/smashes = list()
+	///List of mobs with ability to see the smashes
+	var/list/targets = list()
```

Comment:
Clear these lists on destruction and handle the event of them being destroyed before the datum.

Reply:
if the datum is destroyed, something went really fucky, since that datum is NOT suppoed so to be destroyed

Reply:
What if the targets get destroyed?